mysqli::real_connect may be called with null $flags

### Bug Report

<!-- Fill in the relevant information below to help triage your issue. -->

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 3.1

#### Summary

In PHP 8.1 you get a deprecation when `mysqli::real_connect` is called with null $flags:
`
mysqli::real_connect(): Passing null to parameter #7 ($flags) of type int is deprecated
`
#### Current behaviour

<!-- What is the current (buggy) behaviour? -->

#### How to reproduce

<!--
Provide steps to reproduce the bug.
If possible, also add a code snippet with relevant configuration, driver/platform information, SQL queries, etc.
Adding a failing Unit or Functional Test would help us a lot - you can submit one in a Pull Request separately, referencing this bug report.
-->

#### Expected behaviour

<!-- What was the expected (correct) behaviour? -->

Neither PHPStan nor Psalm in their current configuration catch this issue but PHPStan level 8 would:
```
$ phpstan a src/Driver/Mysqli/Connection.php
Note: Using configuration file phpstan.neon.dist.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------- 
  Line   Connection.php                                                                     
 ------ ----------------------------------------------------------------------------------- 
  61     Parameter #7 $flags of method mysqli::real_connect() expects int, int|null given.  
 ------ ----------------------------------------------------------------------------------- 
                                                           
 [ERROR] Found 1 error

--- a/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
+++ b/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
@@ -68,7 +68,7 @@ class MysqliConnection implements ConnectionInterface, PingableConnection, Serve
         $socket = $params['unix_socket'] ?? ini_get('mysqli.default_socket');
         $dbname = $params['dbname'] ?? null;
 
-        $flags = $driverOptions[static::OPTION_FLAGS] ?? null;
+        $flags = $driverOptions[static::OPTION_FLAGS] ?? 0;
 
         $conn = mysqli_init();
         assert($conn !== false);

